src/plugins/intel_gpu/tests/functional/subgraph_tests/condition.cpp

```diff
@@ -0,0 +1,591 @@
+    InferenceEngine::Precision::FP32,
+    // InferenceEngine::Precision::FP16
+};
+
+const std::vector<CondShapeParams> dynamicInputShapes = {
+    {
+        ov::test::InputShape(ov::PartialShape({1, 1, -1, -1}), {{1, 1, 8, 4}, {1, 1, 24, 16}, {1, 1, 64, 32}, {1, 1, 8, 4}, {1, 1, 24, 16}}),
+        ov::test::InputShape(ov::PartialShape({1}), {{1}, {1}, {1}, {1}, {1}})
+    }
+};
+
+const std::vector<InnerBodyTypeParams> innerBodyTypes = {
+    {
+        InnerBodyGenerator::InnerBodyType::Type01,
+        InnerBodyGenerator::InnerBodyType::Type02
+    },
+    {
+        InnerBodyGenerator::InnerBodyType::Type02,
+        InnerBodyGenerator::InnerBodyType::Type03
+    },
+    {
+        InnerBodyGenerator::InnerBodyType::Type04,
+        InnerBodyGenerator::InnerBodyType::Type03
+    }
+};
+
+const std::vector<InnerBodyTypeParams> innerBodyTypes_for_constant = {
+    {
+        InnerBodyGenerator::InnerBodyType::Type02,
+        InnerBodyGenerator::InnerBodyType::Type03
+    },
+    {
+        InnerBodyGenerator::InnerBodyType::Type04,
+        InnerBodyGenerator::InnerBodyType::Type03
+    }
+};
+
+const std::vector<TestModelGenerator::CondTypes> condTypes = {
+    TestModelGenerator::CondTypes::PARAM,
+    TestModelGenerator::CondTypes::NODE
+};
+
+const std::vector<bool> condExecutionValues = {
+    true, false
+};
+
+INSTANTIATE_TEST_SUITE_P(smoke_ConditionGPUTest_dynamic01, ConditionLayerGPUTest,
+                testing::Combine(
+                    testing::ValuesIn(dynamicInputShapes),                          // input shapes
+                    testing::ValuesIn(innerBodyTypes),                              // inner body type
+                    testing::ValuesIn(netPrecisions),                               // network precision
+                    testing::ValuesIn(condTypes),                                   // cond type
+                    testing::ValuesIn(condExecutionValues),                         // cond execution value
+                    testing::Values<std::string>(CommonTestUtils::DEVICE_GPU)),     // device type
+                ConditionLayerGPUTest::getTestCaseName);
+
+
+INSTANTIATE_TEST_SUITE_P(smoke_ConditionGPUTest_dynamic02, ConditionLayerGPUTest,
+                testing::Combine(
+                    testing::ValuesIn(dynamicInputShapes),                          // input shapes
+                    testing::ValuesIn(innerBodyTypes_for_constant),                 // inner body type
+                    testing::ValuesIn(netPrecisions),                               // network precision
+                    testing::ValuesIn({TestModelGenerator::CondTypes::CONSTANT}),   // cond type
+                    testing::ValuesIn(condExecutionValues),                         // cond execution value
+                    testing::Values<std::string>(CommonTestUtils::DEVICE_GPU)),     // device type
+                ConditionLayerGPUTest::getTestCaseName);
```

Comment:
Could we have a nested branches' teest?
also please add other dimensions' tests  (D=3 && D>=5) examples too.

Reply:
I added more test cases including other dimension (D=3 && D=5) and nested if case